ServerCore/Pages/Threads/PuzzleThreads.cshtml.cs

```diff
@@ -163,6 +168,19 @@ public async Task<IActionResult> OnGetAsync(
                     .ToList();
             }
 
+            ILookup<int, string> puzzleAuthors = (await (from author in _context.PuzzleAuthors
+                                                         where author.Puzzle.Event == Event
+                                                         select author).ToListAsync()).ToLookup(author => author.PuzzleID, author => author.Author.Name);
+
+            AuthorsForPuzzleID = new Dictionary<int, string>();
+
+            foreach (var message in LatestMessagesFromEachThread)
+            { 
+                IEnumerable<string> authorsForPuzzle = puzzleAuthors[message.PuzzleID ?? 0];
```

Comment:
Two things about the safety-checking here:

1. Safety checking is great!
2. Better to just skip messages that have no PuzzleID than to assign them to the wrong puzzle.

Reply:
Great point, I like that!  Skipping definitely seems like the better choice.  I'll make this update pending the int? / int conversation below.